### PR TITLE
CORGI-504: App interface lookup

### PR DESCRIPTION
**Pull request by mprpic**
This interface uses prod-def metadata to extract component composition of
every managed service by listing its source and container image repos.
This metadata can then be used to manifest each component of each service.

### CODE REVIEW

**Comment by mprpic** [comment]
@RedHatProductSecurity/corgi-devs Please review.

File: corgi/collectors/app_interface.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            component_name = component["name"]
+            subcomponent_data: defaultdict = defaultdict(dict)
+
+            quay_repos_data = component.get("quayRepos")
+            if quay_repos_data:
+                for org_repos in quay_repos_data:
+                    # Ignore container images in other registries since we assume all images are
+                    # in Quay.io.
+                    if org_repos["org"]["instance"]["url"] != "quay.io":
+                        logger.info(
```
Comment (juspence) [inline corgi/collectors/app_interface.py]: Should this be a warning?

**Review by juspence — APPROVED** [review]
My understanding of this code is very shallow. It might help to have a docstring for the class and the method explaining what they're doing and how this is used. Or maybe a test with sample data would give a better idea of what we're expecting to receive, do, and return. But overall LGTM.

**Comment by jasinner** [comment]
> My understanding of this code is very shallow. It might help to have a docstring for the class and the method explaining what they're doing and how this is used. Or maybe a test with sample data would give a better idea of what we're expecting to receive, do, and return. But overall LGTM.

I agree, it would be nice to have some tests with sample data.

File: corgi/collectors/app_interface.py (hunk 1 above)
Comment (mprpic) [reply corgi/collectors/app_interface.py]: Sure, why not. Fixed!

**Comment by mprpic** [comment]
@jasinner @juspence Ack on the test. I'll add one in the next MR (CORGI-505) that will add the Celery task to actually fetch the manifesting data, so that I don't have to bother mocking the ProductStream calls in the test specific to this task. There isn't much to test here though, it's just shuffling data around though I guess there is value in having an example response somewhere in the tests.

**Pull request merged**